include/zephyr/usb/usb_device.h

```diff
@@ -446,6 +447,19 @@ int usb_wakeup_request(void);
  */
 bool usb_get_remote_wakeup_status(void);
 
+/**
+ * @brief Initializes the USB Device stack from the Application
+ *
+ * This function exists to provide a mechanism for the application
+ * to initialize the USB device stack. It should be called only from
+ * the application itself. USB_DEVICE_INITIALIZE_BY_APP should be enabled
+ * to allow for this function to be used.
+ *
+ * @return 0 on success, negative errno code on fail
+ */
+int usb_device_init_app(void);
```

Comment:
We will not expand API, which we want to eliminate. There is likely easier way to archive proposed functionality. Please open an enhancement request, motivation is not clear, looks like the commit message got some things mixed up.

Reply:
The reason for the existence of this function is because usb_device_init function is static so cannot be accessed externally. If you are OK with changing the usb_device_init function to be accessible, then this function is not needed.

The title of this is a feature. However, this is also a fix - the usb stack serial callback feature does not work properly because it runs early and therefore cannot work in certain situations when it depends on data. For example, when loading a serial number from hardware. There are other cases, where the USB stack needs to wait for some HW to initialize. 

I will add that the documentation for the stack serial callback is very lacking - no developer would know when the callback actually gets called, leading to issues when trying to use it

Reply:
What is "the usb stack serial callback feature"? Why do you think this is a fix? If it is about hwinfo, it works fine on many platforms I use to test USB support.